independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -151,26 +164,80 @@ public void handle(Throwable failure) {
                             }
                         });
             }
+        }, new Consumer<>() {
+            @Override
+            public void accept(Throwable event) {
+                if (event instanceof IOException) {
+                    ProcessingException throwable = new ProcessingException(event);
+                    reportFinish(0, throwable, requestContext);
+                    requestContext.resume(throwable);
+                } else {
+                    requestContext.resume(event);
+                    reportFinish(0, event, requestContext);
+                }
+            }
         });
     }
 
-    public Future<HttpClientRequest> createRequest(RestClientRequestContext state) {
+    private void reportFinish(long timeInNs, Throwable throwable, RestClientRequestContext requestContext) {
+        ServiceInstance serviceInstance = requestContext.getCallStatsCollector();
+        if (serviceInstance != null) {
+            serviceInstance.recordResult(timeInNs, throwable);
+        }
+    }
+
+    public Uni<HttpClientRequest> createRequest(RestClientRequestContext state) {
         HttpClient httpClient = state.getHttpClient();
         URI uri = state.getUri();
-        boolean isHttps = "https".equals(uri.getScheme());
-        int port = uri.getPort() != -1 ? uri.getPort() : (isHttps ? 443 : 80);
-        RequestOptions requestOptions = new RequestOptions();
-        requestOptions.setHost(uri.getHost());
-        requestOptions.setPort(port);
-        requestOptions.setMethod(HttpMethod.valueOf(state.getHttpMethod()));
-        requestOptions.setURI(uri.getPath() + (uri.getQuery() == null ? "" : "?" + uri.getQuery()));
-        requestOptions.setFollowRedirects(followRedirects);
-        requestOptions.setSsl(isHttps);
         Object readTimeout = state.getConfiguration().getProperty(QuarkusRestClientProperties.READ_TIMEOUT);
-        if (readTimeout instanceof Long) {
-            requestOptions.setTimeout((Long) readTimeout);
+        Uni<RequestOptions> requestOptions;
+        if (uri.getScheme().startsWith("stork")) {
```

Comment:
Is there no constant somewhere in the Stock API we could use?

Reply:
I don't think there is but we sure can add one :)

Reply:
done